crates/assemble/src/bundle.rs

```diff
@@ -50,27 +57,96 @@ pub fn bundled_schema(
             "$defs": Value::Object(dependencies.collect()),
             "$ref": schema.to_string(),
         })
-    } else {
+    } else if dependencies.len() != 0 {
         // This is a reference to the root schema document, which is always the
         // first transitive dependency. Use it as the root of the bundle.
         let mut root = dependencies.next().unwrap().1;
 
         // If there are additional dependencies, merge them into $def's of the root.
-        let to_add: Vec<_> = dependencies.collect();
-        if !to_add.is_empty() {
-            if let Some(defs) = root
-                .as_object_mut()
+        if dependencies.len() != 0 {
```

Comment:
it's not, because we took an item from the iterator already, and it may now be empty if it originally only had one item.

this avoids creating the `$defs` property unnecessarily.